diesel/src/pg/types/date_and_time/std_time.rs

```diff
@@ -57,6 +57,19 @@ fn duration_to_usecs(duration: Duration) -> u64 {
     seconds + u64::from(subseconds)
 }
 
+impl ToSql<sql_types::Interval, Pg> for Duration {
+    fn to_sql<W: Write>(&self, out: &mut Output<W, Pg>) -> serialize::Result {
+        use dsl::IntervalDsl;
+        let seconds = self.as_secs() as i64;
+        let pg_interval = seconds
+            .checked_mul(1000 * 1000)
+            .and_then(|v| v.checked_add(i64::from(self.subsec_nanos() / NANO_PER_USEC)))
+            .map(|v| v.microseconds())
+            .unwrap_or_else(|| seconds.seconds());
```

Comment:
On the one hand I don't like that we're silently losing precision rather than returning an error on overflow. On the other hand this only occurs for durations greater than thousands of years so I don't really care.